src/org/labkey/targetedms/query/SamplePivotCustomizer.java

```diff
@@ -33,92 +31,157 @@
-    public void customize(TableInfo tableInfo)
+    private Integer getRunId()
     {
-        if (HttpView.hasCurrentView())
-        {
-            ViewContext context = HttpView.currentContext();
+        ViewContext context = HttpView.currentContext();
+
+        ActionURL filterURL = context.getActionURL();
 
-            ActionURL filterURL;
+        boolean getQueryDetails = filterURL.getAction().equalsIgnoreCase("getQueryDetails");
 
-            if (context.getActionURL().getAction().equalsIgnoreCase("getQueryDetails"))
+        if (getQueryDetails)
+        {
+            // When getting the query metadata to customize the view, the Run ID parameter doesn't get propagated to
+            // the current URL. Look for it from the referrer URL
+            String referer = context.getRequest() != null ? context.getRequest().getHeader("Referer") : null;
+            filterURL = referer == null ? context.getActionURL() : new ActionURL(referer);
+        }
+
+        if (_urlParameter != null)
+        {
+            String paramValue = filterURL.getParameter(_urlParameter);
+            try
             {
-                // When getting the query metadata to customize the view, the Run ID parameter doesn't get propagated to
-                // the current URL. Look for it from the referrer URL
-                String referer = context.getRequest().getHeader("Referer");
-                filterURL = referer == null ? context.getActionURL() : new ActionURL(referer);
+                return Integer.parseInt(paramValue);
             }
-            else
+            catch (NumberFormatException ignored) {}
+        }
+
+        if (!getQueryDetails)
+        {
+            // Grab any filters that have been applied. Use QuerySettings, which will find the right values whether it's
+            // a GET or POST
+            QuerySettings settings = new QuerySettings(context, "query");
+            filterURL = settings.getSortFilterURL();
+        }
+
+        SimpleFilter filter = new SimpleFilter(filterURL, "query");
+        for (SimpleFilter.FilterClause clause : filter.getClauses())
+        {
+            // Look for an equals filter on the RunId column that has a value specified
+            if (clause.getFieldKeys().contains(_runIdField) &&
+                    clause instanceof CompareType.CompareClause &&
+                    ((CompareType.CompareClause) clause).getCompareType() == CompareType.EQUAL &&
+                    clause.getParamVals().length == 1 &&
+                    clause.getParamVals()[0] != null)
             {
-                // Grab any filters that have been applied. Use QuerySettings, which will find the right values whether it's
-                // a GET or POST
-                QuerySettings settings = new QuerySettings(context, "query");
-                filterURL = settings.getSortFilterURL();
+                try
+                {
+                    // Get the sample names associated with that run
```

Comment:
I am confused with the comment here.

Reply:
Fixed. Agreed that it didn't make sense where it landed after the refactor